internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -557,40 +354,74 @@ func (r *ForeignClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&discoveryv1alpha1.ForeignCluster{}).
 		Owns(&advtypes.Advertisement{}).
 		Owns(&discoveryv1alpha1.PeeringRequest{}).
+		Owns(&discoveryv1alpha1.ResourceRequest{}).
 		Complete(r)
 }
 
+func (r *ForeignClusterReconciler) checkPeeringStatus(ctx context.Context,
+	foreignCluster *discoveryv1alpha1.ForeignCluster) error {
+	var outgoingResourceRequestList discoveryv1alpha1.ResourceRequestList
+	remoteClusterID := foreignCluster.Spec.ClusterIdentity.ClusterID
+	localNamespace := foreignCluster.Status.TenantControlNamespace.Local
+	if err := r.Client.List(ctx, &outgoingResourceRequestList, client.MatchingLabels(resourceRequestLabels(remoteClusterID)),
+		client.InNamespace(localNamespace)); err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	var incomingResourceRequestList discoveryv1alpha1.ResourceRequestList
+	if err := r.Client.List(ctx, &incomingResourceRequestList, client.HasLabels{
+		crdreplicator.ReplicationStatuslabel}, client.MatchingLabels{
+		crdreplicator.RemoteLabelSelector: remoteClusterID,
+	}); err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	if len(outgoingResourceRequestList.Items) > 1 {
+		err := fmt.Errorf("[%v] more than one outgoing resource request found in namespace %v", remoteClusterID, localNamespace)
+		klog.Error(err)
+		return err
+	}
```

Comment:
Probably you could move this check to the `getPeeringPhase` function, using a switch statement